CONTRIBUTING.md

```diff
@@ -56,37 +55,188 @@ Please include the following with each issue:
+
+Here is a sample [changelog](https://github.com/microsoft/rushstack/blob/master/apps/rush/CHANGELOG.md) to demonstrate the level of detail expected.
+
+---
+
+## Contributing Guidelines
+
+AppUI contains a large amount of legacy code. As such we are in the process of standardizing, reorganizing, and optimizing AppUI.
+
+Before contributing to AppUI please consult [the iTwinUI styling guide](https://github.com/iTwin/iTwinUI/blob/main/STYLEGUIDE.md)
+
+When contributing to AppUI please keep in mind the following guidelines.
+
+### Application Architecture
+
+Our `ui` folder contains all our packages which are organized as following:
+
+- `appui-react` should contain exclusively layouts. These are the largest and most complicated components in AppUI.
+- `components-react` should contain exclusively widgets: smaller components that are used inside `appui-react`. Ideally should largely be combinations of iTwinUI components.
+- `core-react` marked to be decrepit at some point. Currently, houses the smallest components inside AppUI. Ideally, all of these components would be replaced with the latest iTwinUI components and the entire core-react could be deleted. Rely on this as little as possible.
+- `imodel-components-react/` contains components that require a 3D environment. Used exclusively for iModels.
+
+### Common issues to solve
+
+Common fixes are we looking to solve include, but ar e not limited to:
+
+- Replacing `core-react` components with iTwinUI components
+- Deleting Snapshot tests
+- Migrating to functional from class based React Components
+
+### Conventions
+
+Please follow these conventions when contributing to AppUI
+
+- Be sure that your branch is up to date with the master branch (i.e. `git merge origin/master`)
+- Classnames should follow this formula: [abrv. of module name] + [folder] + [component]. For example, when working inside the Message Center module on MessageCenterField on the dialog, one would use `”mc-messagecenterfield-dialog”`
+- All components should import a single SCSS file with a matching name, e.g. `Message.tsx` importing `Message.scss`. SCSS files should be siblings of their relative component files.
+
+---
+
+## Contributor License Agreement (CLA)
 
 A [Contribution License Agreement with Bentley](https://gist.github.com/imodeljs-admin/9a071844d3a8d420092b5cf360e978ca) must be signed before your contributions will be accepted. Upon opening a pull request, you will be prompted to use [cla-assistant](https://cla-assistant.io/) for a one-time acceptance applicable for all Bentley projects.
 You can read more about [Contributor License Agreements](https://en.wikipedia.org/wiki/Contributor_License_Agreement) on Wikipedia.
 
-### Pull Requests
+---
 
-All submissions go through a review process.
-We use GitHub pull requests for this purpose.
-Consult [GitHub Help](https://help.github.com/articles/about-pull-requests/) for more information on using pull requests.
+## Testing options
```

Comment:
Nitpick: we should probably make casing consistent. Now in some categories all words start with a capital letter and in other categories only the first word starts with a capital letter. @GerardasB which method should be used?

Reply:
Good point, I think we can keep only first word capitalized.